src/components/display/PhoneInput.tsx

```diff
@@ -0,0 +1,87 @@
+import * as React from "react"
+
+import {
+  InputGroup,
+  InputLeftAddon,
+  Input,
+  InputRightAddon,
+  Select,
+} from "@chakra-ui/react"
+
+export interface PhoneInputProps {
+  onChange: (phoneNumber: string) => void
+  defaultCountry?: "AU" | "US"
+}
+
+export const PhoneInput = (props: PhoneInputProps) => {
+  const { onChange, defaultCountry = "AU" } = props
+
+  const [country, setCountry] = React.useState(defaultCountry)
+  const [number, setNumber] = React.useState()
+
+  const isAustralia = country === "AU"
+
+  const onCountrySelectChange = React.useCallback(
+    (event) => {
+      setCountry(event.target.value)
+    },
+    [setCountry]
+  )
+
+  const onInputChange = React.useCallback(
```

Comment:
this callback will already need to change a little bit given the above feedback. however, there's some more changes which I think will be useful.

instead of allowing the user to enter anything and then slicing up the value depending on which country is currently selected, we should disallow them from entering anything other than a number. in future I think we should format the visual value with spaces, too, but let's start simple for right now.

we also shouldn't worry about enforcing a particular format just yet, we'll do that in the next phase and delegate to an established library like `libphonenumber-js` instead of reinventing the wheel. 

here's how I think this callback should behave

- [ ] Allow only numeric characters to be entered. I _think_ this might be possible by using [event.preventDefault](https://developer.mozilla.org/en-US/docs/Web/API/Event/preventDefault) or similar, but I'm not 100% sure.
- [ ] Prefix the entered number with the current country's calling code, and..
- [ ] Create a new `event` whose `target.value` contains the formatted number and pass tat through to `inputProps.onChange` (we want to pass up the entire DOM event, not just a string)
- [ ] No need to mutate any state

example for task 3..

```typescript
const updatedEvent = { ...event, { target: { ...event.target, value: phoneNumber } } }
inputProps.onChange(updatedEvent)
// ^ this way parent components will only ever see the formatted/decorated number
// instead of the user's raw input
```